src/GoTrueClient.ts

```diff
@@ -503,10 +551,10 @@ export default class GoTrueClient {
       const timeNow = Math.round(Date.now() / 1000)
       const expires_at = timeNow + parseInt(expires_in)
 
-      const { data, error } = await this.api.getUser(access_token)
+      const { data, error } = await this.getUser()
```

Comment:
@kangmingtay doesn't this change break magic links, oauth callbacks et all? Usually there shouldn't be a session in storage at this point. If there is a session in storage we can't be sure it is a session for the same user as the access_token in the url.

Reply:
yeah you're right, we're putting up a fix here (https://github.com/supabase/gotrue-js/pull/384) - thanks for pointing this out, many changes were made here and the goal was to simplify the structure of the code base as much as possible and this was unfortunately overlooked :/